### PR TITLE
 Add a provider field fixes #77

### LINKED ISSUE #77: Add a provider field
Parse the URL into a top level name which removes 'www' and any TLD, ie: 'https://www.amazon.ca/product?id=123' becomes 'amazon'

### PR DESCRIPTION
**Pull request by jaredkerim**

### CODE REVIEW

**Comment by jaredkerim** [comment]
Allowed support for subdomains so 'sports.bing.com' will render as 'sports bing' which I think is a reasonable compromise.

File: parser.js [hunk 1]
```diff
@@ -11,6 +11,16 @@ function makeUrlAbsolute(base, relative) {
   return relative;
 }
 
+function getProvider(url) {
+  return urlparse.parse(url)
+    .hostname
+    .replace(/www[a-zA-Z0-9]*\./, '')
+    .replace('co.', '')
```
Comment (pdehaan) [inline parser.js]: The special casing of this one particular second-level TLD is mildly concerning.
Not sure how many others (if any), we should bother to handle: https://en.wikipedia.org/wiki/Second-level_domain

But probably can call it "good enough" for now, and address in the future if we notice weird results.

```js
function getProvider(hostname) {
  const out = hostname
    .replace(/www[a-zA-Z0-9]*\./, '')
    .replace('co.', '')
    .split('.')
    .slice(0, -1)
    .join(' ');
  console.log(out);
}

getProvider('yahoo.com'); // "yahoo"
getProvider('www.reddit.com'); // "reddit"
getProvider('sports.bing.com'); // "sports bing"
getProvider('www.bbc.co.uk'); // "bbc"
getProvider('www.cnn.com'); // "cnn"
getProvider('aeon.co'); // "aeon"
getProvider('mobile.nytimes.com'); // "mobile nytimes"
getProvider('www1.foobar.ca'); // "foobar"
getProvider('mobile.www.barfoo.ninja'); // "mobile barfoo" // <== removes "www." from middle of URL.
getProvider('www.school.k12.il'); // "school k12"
```

Comment (pdehaan) [inline parser.js]: Other interesting results from random Reddit /r/worldnews results:

```js
getProvider('www.businessinsider.com.au'); // "businessinsider com"
getProvider('abcnews.go.com'); // "abcnews go"
getProvider('timesofindia.indiatimes.com'); // "timesofindia indiatimes"
getProvider('www.abc.net.au'); // "abc net"
getProvider('bigstory.ap.org'); // "bigstory ap"
```

**Comment by pdehaan** [comment]
Last one, here's the sorted and deduped results of the first 5 pages of /r/worldnews through your provider function (where the provider is more than 1 word):

```js
[ 'abc net',
  'abcnews go',
  'bigstory ap',
  'businessinsider com',
  'dailystar com',
  'economictimes indiatimes',
  'edition cnn',
  'globalnation inquirer',
  'm ndtv',
  'mobile reuters',
  'motherboard vice',
  'nakedsecurity sophos',
  'news sky',
  'news vice',
  'timesofindia indiatimes' ]
```

```js
const urlparse = require('url');
const { fetchSubreddit, domainReducer } = require('reddit-as-json');

fetchSubreddit('worldnews', 5)
  .then(domainReducer)
  .then(({data}) => data.map((domain) => getProvider(`https://${domain.name}`)))
  .then((data) => {
    return Object.keys(data.reduce((prev, curr) => {
      prev[curr] = true;
      return prev;
    }, {})).sort();
  })
  .then((data) => data.filter((provider) => provider.split(' ').length > 1))
  .then((data) => console.log(data))
  .catch((err) => console.error(err));

function getProvider(url) {
  return urlparse.parse(url)
    .hostname
    .replace(/www[a-zA-Z0-9]*\./, '')
    .replace('co.', '')
    .split('.')
    .slice(0, -1)
    .join(' ');
}
```

**Pull request merged**